bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/DataClientVeneerApi.java

```diff
@@ -229,8 +249,97 @@ protected void onCompleteImpl() {
   }
 
   /** wraps {@link ServerStream} onto HBase {@link ResultScanner}. */
-  private static class RowResultScanner extends AbstractClientScanner {
+  static class PaginatedRowResultScanner extends AbstractClientScanner {
+    // Percentage of max number of rows allowed in the buffer
+    private static final double WATERMARK_PERCENTAGE = .1;
+    private static final int MIN_BYTE_BUFFER_SIZE = 100 * 1024 * 1024;
+    private static final double DEFAULT_BYTE_LIMIT_PERCENTAGE = .1;
+    private static final RowResultAdapter RESULT_ADAPTER = new RowResultAdapter();
 
+    private final Meter scannerResultMeter =
+        BigtableClientMetrics.meter(BigtableClientMetrics.MetricLevel.Info, "scanner.results");
+    private final Timer scannerResultTimer =
+        BigtableClientMetrics.timer(
+            BigtableClientMetrics.MetricLevel.Debug, "scanner.results.latency");
```

Comment:
Im not sure we want to expand the drop wizard style metrics. I'm planning on dropping this soon

Reply:
Agreed not to change anything